webserver.md

```diff
@@ -0,0 +1,280 @@
+      // If there is a request containing the string 'leds' call a function, toggleLED
+      toggleLED(urlParts.pathname, request, response);
+    } else {
+      // All other request will call a function, showIndex
+      showIndex(urlParts.pathname, request, response);
+    }
+  });
+
+  // Stays the same
+  server.listen(8080);
+
+  // Stays the same
+  console.log('Server running at http://192.168.1.101:8080/');
+
+  // Respond to the request with our index.html page
+  function showIndex (url, request, response) {
+    // Create a response header telling the browser to expect html
+    response.writeHead(200, {"Content-Type": "text/html"});
+
+    // Use fs to read in index.html
+    fs.readFile(__dirname + '/index.html', function (err, content) {
+      // If there was an error, throw to stop code execution
+      if (err) {
+        throw err;
+      }
+
+      // Serve the content of index.html read in by fs.readFile
+      response.end(content);
+    });
+  }
+
+  // Toggle the led specified in the url and respond with its state
+  function toggleLED (url, request, response) {
+    // Create a regular expression to find the number at the end of the url
+    var indexRegex = /(\d)$/;
+
+    // Capture the number, returns an array
+    var result = indexRegex.exec(url);
+
+    // Grab the captured result from the array
+    var index = result[1];
+
+    // Use the index to refence the correct LED
+    var led = tessel.led[index];
+
+    // Toggle the state of the led and call the callback after that's done
+    led.toggle(function (err) {
+      if (err) {
+        // Log the error, send back a 500 (internal server error) response to the client
+        console.log(err);
+        response.writeHead(500, {"Content-Type": "application/json"});
+        response.end(JSON.stringify({error: err}));
+      } else {
+        // The led was successfully toggled, respond with the state of the toggled led using led.isOn
+        response.writeHead(200, {"Content-Type": "application/json"});
+        response.end(JSON.stringify({on: led.isOn}));
+      }
+    });
+  }
+{% endhighlight %}
+
+In order to make code pushing more efficient, Tessel only pushes the entry point file and its Node dependencies by default. Since index.html is not included in this default push, we'll need to explicitly require it with a `.tesselinclude` file.
+
+Create a new file called `.tesselinclude` and copy and paste the following:
+
+{% highlight javascript %}
+index.html
+{% endhighlight %}
+
+Finally, let's fire up our server again by running:
+
+`t2 run webserver.js`
+
+</div>
+
+<div class="small-12 columns">
+Once the server is running, connect to the URL in the web browser. 
+
+Click on the buttons to toggle the lights.
+</div>
+
+<div class="small-12 columns">
+Here is a demo video:
+
+<video src="https://dl.dropboxusercontent.com/u/74986127/tessel-router-demo.mp4" controls loop class="small-12" >Video not available at the moment.</video>
```

Comment:
It ended up being too low resolution. Plus a video gives people the ability to fullscreen and control the video playback. Since this demo has a few moving parts to it, I think the video will be a better preview.